### ISSUE TITLE
Implementing ControlSystem in ModelingToolkit

### ISSUE THREAD

**Issue by anujdave23** [SciML/ModelingToolkit.jl#749]
Hello fellas,
I am trying to develop one simple example of controlsystem for the optimal control of any system but getting some error.
I tried Moon Lander problem.But i didnt understand the error .
Please to code that i have developed and give me some feedback to run it successfully
![moon_lander](https://user-images.githubusercontent.com/77669053/105176660-cdd2e480-5b25-11eb-94db-aa544c1c1286.jpg)
![controlsystem_moon_lander](https://user-images.githubusercontent.com/77669053/105176666-d0cdd500-5b25-11eb-867b-0d48f557a54d.jpg)

**Comment by ChrisRackauckas** [comment]
Can you post this as code?

**Comment by anujdave23** [comment]
> Can you post this as code?

using ModelingToolkit,QuadGK

@variables t x(t) v(t) h(t)
@parameters p , g
D = Differential(t)

loss = quadgk(u,0,p)
eqs = [
    D(h) ~ v 
    D(v) ~ -g + u
]

sys = ControlSystem(loss,eqs,t,[h,v],[u],[p,g])
dt = 0.1
tspan = (0.0,1.0)
sys = runge_kutta_discretize(sys,dt,tspan)

u0 = rand(112) # guess for the state values
prob = OptimizationProblem(sys,u0,[0.1],grad=true)

**Comment by ChrisRackauckas** [comment]
`u` isn't defined. The loss is supposed to be the integrand, so no need to write the integral.

**Comment by anujdave23** [comment]
so if i initiate u(t) as an variable and in loss function only u , will that be fine?

> `u` isn't defined. The loss is supposed to be the integrand, so no need to write the integral.

**Comment by ChrisRackauckas** [comment]
Use the test as an example: https://github.com/SciML/ModelingToolkit.jl/blob/master/test/controlsystem.jl. We're probably still a few months away from having a full blown tutorial there though.

**Comment by anujdave23** [comment]
> Use the test as an example: https://github.com/SciML/ModelingToolkit.jl/blob/master/test/controlsystem.jl. We're probably still a few months away from having a full blown tutorial there though.

Actually I am a Master Student and this is my embedded seminar topic, so i need to develop a simple optimal control system using controlsystem and i can't copy and paste your code. So i tried to develop a system but i am not getting what the error is

**Comment by ChrisRackauckas** [comment]
I'm saying right now the best documentation is the test, so use the test as an example of what works.

**Comment by anujdave23** [comment]
Okay thanks for your information, i also have a doubt about equality_constarints and inequality_contraints in optimizationsystem of MoodelingToolkit . Is it possible to apply the condition if yes how? Can you please guide me through this?

**Comment by ChrisRackauckas** [comment]
That's not available yet. ControlSystem is pretty unfinished right now and I don't expect it to be completed for another couple months.

**Comment by ChrisRackauckas** [comment]
I'm not sure OptimizationSystem will do it all, but you can directly modify the OptimizationProblem.
